plugins/package-managers/gleam/src/main/kotlin/Gleam.kt

```diff
@@ -84,18 +84,14 @@ class Gleam internal constructor(
         definitionFiles: List<File>,
         analyzerConfig: AnalyzerConfiguration
     ): List<File> {
-        val projectFiles = definitionFiles.toMutableList()
-
         // Ignore definition files from build directories that reside next to other definition files,
         // to avoid dependency gleam.toml files from being recognized as projects.
-        var index = 0
-        while (index < projectFiles.size - 1) {
-            val projectFile = projectFiles[index++]
-            val buildDir = projectFile.resolveSibling("build")
-            projectFiles.subList(index, projectFiles.size).removeAll { it.startsWith(buildDir) }
+        val buildDirs = definitionFiles.map { it.resolveSibling("build") }
```

Comment:
Note that this code got inspired from

https://github.com/oss-review-toolkit/ort/blob/60316b797569f05fc7835f6e2f3b2bb72340f71b/plugins/package-managers/composer/src/main/kotlin/Composer.kt#L116-L123

So if any simplification is done here, it should probably be done here, too.

But I have a hunch that the algorithm (at least for Composer) was deliberately written this was. I recall a discussion we have [here](https://github.com/oss-review-toolkit/ort/pull/9864), but I need to revisit the details.

Reply:
Note that composer has a dedicated test for that function (and gleam the same).